classes/Topic.php

```diff
@@ -0,0 +1,339 @@
+<?php
+/**
+ * Topic
+ *
+ * @package TheyWorkForYou
+ */
+
+namespace MySociety\TheyWorkForYou;
+
+class Topic {
+
+    /**
+     * DB handle
+     */
+    private $db;
+
+    private $raw;
+    private $id;
+    private $title;
+    private $slug;
+    private $description;
+    private $image;
+    private $search_string;
+    private $front_page;
+    private $actions;
+
+    /**
+     * Constructor
+     *
+     */
+
+    public function __construct($data = NULL)
+    {
+        $this->db = new \ParlDB;
+
+        if (is_null($data)) {
+          return;
+        }
+
+        $this->raw = $data;
+        if (isset($data['id'])) {
+            $this->id = $data['id'];
+        }
+        $this->title = $data['title'];
+        $this->slug = $data['slug'];
+        $this->description = $data['description'];
+        $this->image = $data['image'];
+        if (isset($data['search_string'])) {
+            $this->search_string = $data['search_string'];
+        }
+        if (isset($data['front_page'])) {
+            $this->front_page = $data['front_page'];
+        }
+
+        $this->raw['actions'] = array();
+    }
+
+
+    function data() {
```

Comment:
I guess my other concern with `data` is that it is not updated when any `set` functions are called. It looks like the code *either* changes things or displays and uses `data`, so it's okay, but you could see in future something perhaps setting and fetching data assuming it would be up to date.